### DISCUSSION TITLE
Stupid question: can I use LeftWM in a Desktop Environment like KDE?

### DISCUSSION THREAD

**Discussion by bgalvao** [leftwm/leftwm#1071] [Q&A]
Self-explanatory.

I am a complete noob with WMs, and sometimes I don't know whether they work on top of the DE because of being being independent of the DE - like KWins script extensions that support a form of tiling.

Or whether this is like i3, and is a completely different DE.

I feel like [this person](https://www.reddit.com/r/learnprogramming/comments/qie409/where_do_i_write_my_code/) right now. (No pejorative intent towards them is intended).

Please 🐻 with me.

**Comment by VuiMuich** [comment]
Welcome fellow adventurer!

A DE short for Desktop Environment is a whole set of tools to give the user a complete experience. Afaik all of them come with their own WM (Window Manager) as this is a necessary part of a working x11 session.
As there can only ever run one WM at a time it is not possible to run leftwm "on top" of a KDE session but of course you can have both installed and just select to which one you want to log in. While many of the tools (especially utilities like tray applications or apps that directly communicate with the WM etc.) are pretty much tied to the other applications of the suite some might still work with other WMs (e.g. xfce4-clipboard). For testing purposes `Xephyr` can simulate a independent x-session so any WM can be run in a Window, but this is probably not desirable as a actual working environment.

This may not be that indepth as this incredible top answer in the reddit post you linked, but I hope it still helps you getting further into the adventure. And if anything is still unclear, just ask.

**Comment by 0323pin** [comment]
Although what @VuiMuich is true, it's possible to replace the WM on a DE.

Haven't tried to do it in KDE and replace `kwin` with LeftWM but, here's [LeftWM as the WM of LXQt](https://www.reddit.com/r/UsabilityPorn/comments/udrnpa/leftwm_lxqt_tiling_bsd_desklaptop/?utm_source=share&utm_medium=android_app&utm_name=androidcss&utm_term=1&utm_content=share_button)

@VuiMuich thanks for the invitation, accepted 😀

**Reply by VuiMuich** [reply]
Oh, thanks for adding this.
I guess I saw this reddit post, but forgot about it.

Probably lxqt is better suited for hacking a WM replacement then KDE or other tightly knit DEs. I vaguely remember one user who came from a typical "beginner distro" and some parts of the DE were stubbornly starting without a trace in services and xdg autostart or xinit and causing trouble not finding their expected WM.

Regarding the invitation: are you using any IM we could reach out to you outside of github? Most of us others are on discord, but if you don't like to use it, thats understandable and we don't want to force it on you.

**Comment by 0323pin** [comment]
> are you using any IM we could reach out to you outside of github?

Yes, I'm on Matrix but, just drop me an e-mail to start with. My e-mail is public domain, [line 7](http://cvsweb.netbsd.org/bsdweb.cgi/pkgsrc/wm/leftwm/Makefile?rev=1.7&content-type=text/x-cvsweb-markup) of the package Makefile 😉
